From 853b2f4ff2186fee299901f944d35e29cf228451 Mon Sep 17 00:00:00 2001
From: Hauke Mehrtens <hmehrtens@maxlinear.com>
Date: Tue, 12 Oct 2021 18:11:45 +0200
Subject: Improve setting the learning limit

The learning limit was not handled in all functions of the bridge.c file
which should handle it.

The missing part in bridge_override_rtm() broke setting the learning
limit.

Fixes: 11306325e0c9 ("add mac learning limit")
---
 lib/route/link/bridge.c | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/lib/route/link/bridge.c b/lib/route/link/bridge.c
index e0bdbd2..335fdbd 100644
--- a/lib/route/link/bridge.c
+++ b/lib/route/link/bridge.c
@@ -346,7 +346,8 @@ static int bridge_override_rtm(struct rtnl_link *link) {
         if (bd->ce_mask & BRIDGE_ATTR_FLAGS ||
             bd->ce_mask & BRIDGE_ATTR_COST ||
             bd->ce_mask & BRIDGE_ATTR_PRIORITY ||
-            bd->ce_mask & BRIDGE_ATTR_PORT_STATE)
+            bd->ce_mask & BRIDGE_ATTR_PORT_STATE ||
+            bd->ce_mask & BRIDGE_ATTR_LEARNING_LIMIT)
                 return 1;
 
         return 0;
@@ -439,6 +440,9 @@ static void bridge_dump_details(struct rtnl_link *link,
 	if (bd->ce_mask & BRIDGE_ATTR_COST)
 		nl_dump(p, "cost %u ", bd->b_cost);
 
+	if (bd->ce_mask & BRIDGE_ATTR_LEARNING_LIMIT)
+		nl_dump(p, "learning-limit %i ", bd->b_learning_limit);
+
 	if (bd->ce_mask & BRIDGE_ATTR_HWMODE) {
 		char hbuf[32];
 
@@ -471,6 +475,7 @@ static int bridge_compare(struct rtnl_link *_a, struct rtnl_link *_b,
 	diff |= BRIDGE_DIFF(PORT_STATE,	a->b_port_state != b->b_port_state);
 	diff |= BRIDGE_DIFF(PRIORITY, a->b_priority != b->b_priority);
 	diff |= BRIDGE_DIFF(COST, a->b_cost != b->b_cost);
+	diff |= BRIDGE_DIFF(LEARNING_LIMIT, a->b_learning_limit != b->b_learning_limit);
 	diff |= BRIDGE_DIFF(PORT_VLAN, memcmp(&a->vlan_info, &b->vlan_info,
 					      sizeof(struct rtnl_link_bridge_vlan)));
 	diff |= BRIDGE_DIFF(HWMODE, a->b_hwmode != b->b_hwmode);
